src/Microsoft.ML.AutoML/Utils/SplitUtil.cs

```diff
@@ -47,9 +47,12 @@ internal static class SplitUtil
         /// Split the data into a single train/test split.
         /// </summary>
         public static (IDataView trainData, IDataView validationData) TrainValidateSplit(MLContext context, IDataView trainData,
-            string samplingKeyColumn)
+            string samplingKeyColumn, bool shuffle = false)
         {
             var originalColumnNames = trainData.Schema.Select(c => c.Name);
+            if (shuffle)
+                trainData = context.Data.ShuffleRows(trainData);
```

Comment:
Any particular reason to shuffle here? Seems like an odd place to add a shuffle. If for no other reason you're also shuffling the validation split. There's another code path for the dataset splits where the user can provide the train/validation split directly; in this case your shuffle will not run.

I assume you're looking for the ImageClassification trainer to receive shuffled data. It's likely you'll want the shuffle right before your trainer. Alternatively it should be part of the trainer itself (adding it itself) w/ the normal `Shuffle = true` hyperparameter. Perhaps a `RowShufflingTransformer` just before sending data to TF.

Reply:
We need to shuffle otherwise you can train on set of images and test on another set of images that you did not train on, images need to be shuffled before they are fed into the trainer. Think of a dataset that has images of cats followed by images of dogs, now whether you do CV or TrainTest, you will end up training on cats and testing on dogs and get a very poor accuracy.

---
In reply to: [338993257](https://github.com/dotnet/machinelearning/pull/4384#discussion_r338993257) [](ancestors = 338993257)

Reply:
See comment above on the non-need of shuffling for TrainValidate/CV splits: https://github.com/dotnet/machinelearning/pull/4384/files#r339313039